how-it-works/1_protocol/01_peer-to-peer-p2p.subpage.md

```diff
@@ -5,10 +5,73 @@ shareImage: /img/how-it-works/peer-to-peer-p2p.600.jpg
+By every node in the network doing exactly this, every artifact *eventually* propagates through the whole subnet.
+
+## Asynchronous Communication Networks
+
+It is important to note that artifacts reach all subnet nodes eventually, that is, no upper bound on when this happens can be given.
+This important property of an *asynchronous communication network*, which reflects the real world of networks, is crucial and the IC consensus protocol is built with this assumption.
+Clearly, assuming an asynchronous network makes protocols building on top more complex than they would be when assuming a synchronous network.
+
+## Adverts
+
+Ensuring the eventual delivery of broadcasts to all nodes in the subnet is achieved by a node providing a received or protocol-generated message to all its connected peers.
+However, doing so in a naïve way by simply forwarding the message to all peers would result in nodes receiving as many copies of the same message as the node has peers, which unnecessarily consumes networking bandwidth for transmitting duplicate messages and reduces the achieveable throughput of the subnet.
+
+This duplication of delivered artifacts in the naïve approach is mitigated by nodes sending *adverts* for artifacts to their peers instead of sending the artifacts themselves.
+An advert specifies its corresponding artifact, but is a small message only containing the hash of the artifact to unambiguously refer to it and some additional metadata.
+Only adverts for artifacts are sent (pushed) to all its peers by a node.
+A node receiving adverts then requests each artifact only once from one of its peers that has advertised the artifact through an advert.
+For artifacts that are larger than the advert size, this saves bandwidth on the network connections between the peers – the savings grow with increasing size of the artifacts.
+
+## Prioritization of Artifacts
+
+The P2P layer allows the prioritization of artifacts such that the more crucial artifacts are broadcast throughout the subnet nodes more quickly than the others.
+Prioritizing some artifacts over others is important to ensure that the protocol can always make progress and not be starved of network bandwidth for "less important" traffic.
+This principle is well known from traditional networking and applies equally well to a blockchain system.
 
-The Internet Computer blockchain is designed to be secure, reliable, and scalable. Scalability is provided through subnet blockchains that run Canister smart contracts on a selection of nodes. The peer-to-peer layer ("P2P") enables reliable and secure communication between nodes participating in the same subnet. P2P collects and advertises messages from users, as well as from other nodes, so that they will eventually be replicated across all nodes in the same subnet blockchain.
+## Sparse Overlay Network Topology
```

Comment:
I'd omit this part. When we recorded the video, we thought we might do that soon. Looking forward today, I don't think we will do anything like that any time in the foreseeable future.